packages/driver/cypress/integration/commands/actions/selectFile_spec.js

```diff
@@ -0,0 +1,675 @@
+const { _, $ } = Cypress
+
+// Reading and decoding files from an input element would, in the real world,
+// be handled by the application under test, and they would assert on their
+// application state. We want to assert on how selectFile behaves directly
+// though, and getting the files associated with an <input> as strings is
+// handy.
+function getFileContents (subject) {
+  const decoder = new TextDecoder('utf8')
+
+  const fileContents = _.map(subject[0].files, (f) => {
+    return f
+    .arrayBuffer()
+    .then((c) => decoder.decode(c))
+  })
+
+  return Promise.all(fileContents)
+}
+
+describe('src/cy/commands/actions/selectFile', () => {
+  beforeEach(() => {
+    cy.visit('/fixtures/files-form.html')
+    cy.wrap(Buffer.from('foo')).as('foo')
+  })
+
+  context('#selectFile', () => {
+    it('attaches a single file', () => {
+      cy.get('#basic')
+      .selectFile({ contents: '@foo', fileName: 'foo.txt' })
+
+      cy.get('#basic')
+      .then((input) => {
+        expect(input[0].files.length).to.eq(1)
+        expect(input[0].files[0].name).to.eq('foo.txt')
+        expect(input[0].files[0].type).to.eq('')
+        expect(input[0].files[0].lastModified).to.be.closeTo(Date.now(), 1000)
+      })
+
+      cy.get('#basic')
+      .then(getFileContents)
+      .then((contents) => {
+        expect(contents[0]).to.eql('foo')
+      })
+    })
+
+    it('attaches multiple files', () => {
+      cy.get('#multiple')
+      .selectFile([
+        {
+          contents: '@foo',
+          fileName: 'foo.txt',
+        }, {
+          contents: Buffer.from('{"a":"bar"}'),
+          fileName: 'bar.json',
+        },
+        Buffer.from('baz'),
+      ])
+
+      cy.get('#multiple')
+      .should('include.value', 'foo.txt')
+      .then((input) => {
+        expect(input[0].files[0].name).to.eq('foo.txt')
+        expect(input[0].files[1].name).to.eq('bar.json')
+        expect(input[0].files[2].name).to.eq('')
+      })
+
+      cy.get('#multiple')
+      .then(getFileContents)
+      .then((contents) => {
+        expect(contents[0]).to.eq('foo')
+        expect(contents[1]).to.eq('{"a":"bar"}')
+        expect(contents[2]).to.eq('baz')
+      })
+    })
+
+    it('attaches files with custom lastModified', () => {
```

Comment:
```suggestion
    it('selects files with custom lastModified', () => {
```

Reply:
Went through and applied the rest of these as part of the same commit.